Common/cpp/NativeModules/NativeReanimatedModule.h

```diff
@@ -127,9 +131,19 @@ class NativeReanimatedModule : public NativeReanimatedModuleSpec {
       const jsi::Value &commandNameValue,
       const jsi::Value &argsValue);
 
+  jsi::String obtainProp(
```

Comment:
Not really sure why we need all of those functions, why do we have so much of them?
* `getViewProp`
* `obtainProp`
* `obtainPropFromShadowNode`

What's the difference between `getViewProp` and `obtainProp`? I assume that one is used from the RN runtime and the other is used from the UI runtime. Can we somehow unify it so that it's consistent across runtimes? (We can compare `jsi::Runtime &rt` with `uiRuntime_->getJSIRuntime()` to see if we're on the UI runtime.)

What's the difference between `obtainProp` and `obtainPropFromShadowNode`? Can we merge one into another?

Reply:
* `obtainPropFromShadowNode` is a helper function used both in `getViewProp` and `obtainProp`
* The difference between `obtainProp` and `getViewProp` is exactly in runtimes. The current form of these functions was introduced here: https://github.com/software-mansion/react-native-reanimated/pull/4929 We could try unify them, but that would be a vary big change (and a very dangerous), so I wouldn't recommend doing it in this PR

Reply:
I think we can make those functions tied to runtimes for now, and can make them runtime-agnostic in the future, if needed.

Reply:
I would love to see some unification, right now there's too many functions and it's confusing.

Reply:
I agree about the unification, but I think this PR is too important too keep it waiting at function splitting/merging.